src/serviceFacades/sharing.js

```diff
@@ -22,10 +27,192 @@ import callApi from "../common/callApi";
+
+export const unshareApps = ({ appUnsharingRequest }) => {
+    return callApi({
+        endpoint: `/api/apps/unsharing`,
+        method: "POST",
+        body: appUnsharingRequest,
+    });
+};
+
+export const shareAnalyses = ({ analysisSharingRequest }) => {
+    return callApi({
+        endpoint: `/api/analyses/sharing`,
+        method: "POST",
+        body: analysisSharingRequest,
+    });
+};
+
+export const unshareAnalyses = ({ analysisUnsharingRequest }) => {
+    return callApi({
+        endpoint: `/api/analyses/unsharing`,
+        method: "POST",
+        body: analysisUnsharingRequest,
+    });
+};
+
+export const shareTools = ({ toolSharingRequest }) => {
+    return callApi({
+        endpoint: `/api/tools/sharing`,
+        method: "POST",
+        body: toolSharingRequest,
+    });
+};
+
+export const unshareTools = ({ toolUnsharingRequest }) => {
+    return callApi({
+        endpoint: `/api/tools/unsharing`,
+        method: "POST",
+        body: toolUnsharingRequest,
+    });
+};
+
+export const searchSubjects = ({ searchTerm }) => {
+    return callApi({
+        endpoint: `/api/subjects`,
+        params: { search: searchTerm.trim() },
+    });
+};
+
+export const doSharingUpdates = ({ sharing, unsharing }) => {
+    const { data, apps, analyses, tools } = sharing;
+    const {
+        data: dataUnshare,
+        apps: appsUnshare,
+        analyses: analysesUnshare,
+        tools: toolsUnshare,
+    } = unsharing;
+    let promises = [];
+    if (data && data.length > 0) {
+        promises.push(dataSharing({ sharingReq: { sharing: data } }));
+    }
+    if (dataUnshare && dataUnshare.length > 0) {
+        promises.push(
+            dataUnsharing({ unsharingReq: { unsharing: dataUnshare } })
+        );
+    }
+    if (apps && apps.length > 0) {
+        promises.push(shareApps({ appSharingRequest: { sharing: apps } }));
+    }
+    if (appsUnshare && appsUnshare.length > 0) {
+        promises.push(
+            unshareApps({ appUnsharingRequest: { unsharing: appsUnshare } })
+        );
+    }
+    if (analyses && analyses.length > 0) {
+        promises.push(
+            shareAnalyses({ analysisSharingRequest: { sharing: analyses } })
+        );
+    }
+    if (analysesUnshare && analysesUnshare.length > 0) {
+        promises.push(
+            unshareAnalyses({
+                analysisUnsharingRequest: { unsharing: analysesUnshare },
+            })
+        );
+    }
+    if (tools && tools.length > 0) {
+        promises.push(shareTools({ toolSharingRequest: { sharing: tools } }));
+    }
+    if (toolsUnshare && toolsUnshare.length > 0) {
+        promises.push(
+            unshareTools({ toolUnsharingRequest: { unsharing: toolsUnshare } })
+        );
+    }
+
+    return new Promise((resolve, reject) => {
+        Promise.all(promises)
+            .then((values) => {
+                resolve(values);
+            })
+            .catch((error) => {
+                reject(error);
+            });
```

Comment:
I think these lines are unneeded, and the `return new Promise((resolve, reject) => { })` wrapper. next.js uses [bluebird](http://bluebirdjs.com) for the Promise implementation, not that it matters much since it's based on a standard, and what [`Promise.all`](http://bluebirdjs.com/docs/api/promise.all.html) already does is return a promise that... resolves with an array of values or rejects with an error. So this whole thing can just be `return Promise.all(promises)`